File: Templates/Templates/TemplateOne/src/app/core/models/Category.ts
import { Product } from 'src/app/core/models/Product';
import { CategoryProperty } from 'src/app/core/models/CategoryProperty';

export class Category{

  categoryName: string;
  categoryDescription: string;
  productList: Product[];
  subCategories: Category[];
  categoryProperty: CategoryProperty[];
  parentCategoryId: string;
  id: string;
  customerId: string;
  mainCategory: boolean;
  subCategoriesIds: string[];
  productIds: string[];
}

File: Backend/Wiz/DataModels/DataModels/Category.cs
using AbstractModels;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson.Serialization.IdGenerators;
using System;
using System.Collections.Generic;

namespace DataModels
{
    [BsonIgnoreExtraElements]
    public class Category : DataClass
    {
        [BsonIgnoreIfDefault]
        public string CategoryName { get; set; }
        [BsonIgnoreIfDefault]
        public string CategoryDescription { get; set; }
        [BsonIgnoreIfDefault]
        public List<string> ProductIds { get; set; }
        [BsonIgnoreIfDefault]
        public List<Product> Products { get; set; }
        [BsonIgnoreIfDefault]
        public List<string> SubCategoriesIds { get; set; }
        [BsonIgnoreIfDefault]
        public List<Category> SubCategories { get; set; }
        [BsonIgnoreIfDefault]
        public List<CategoryProperty> Properties { get; set; }
        [BsonIgnoreIfDefault]
        [BsonRepresentation(BsonType.ObjectId)]
        public string ParentCategoryId { get; set; }

        public bool MainCategory { get; set; } = false;

    }
}